pkg/providers.bzl

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Packaging related providers."""
+
+
+PackageArtifactInfo = provider(
+    doc = """Metadata about a package artifact.""",
+    fields = {
+        "label": "Label which produced it",
+        "file_name": "The file name of the artifact.",
+    },
+)
+
+PackageNamingInfo = provider(
+    doc = """Metadata about a package artifact.""",
+    fields = {
+        "values": "Dict of name/value pairs",
+    },
+)
```

Comment:
Since this information could also be used to influence things like paths within packages (#210), perhaps this should be named `PackageTemplateInfo` instead?

Reply:
How about just "PackageVariables" - in the same spirt as Make variables https://docs.bazel.build/versions/master/be/make-variables.html

That does not imply either file names or templates, so it is fair to use them anywhere we think it is fit.

Reply:
Not a bad idea, although it wouldn't mesh all that well with the standard that providers are named `Info`.

On further thought, would it make sense to use the builtin [`TemplateVariableInfo`](https://docs.bazel.build/versions/master/skylark/lib/TemplateVariableInfo.html) provider instead?

Reply:
I decided to stick with PackageVariablesInfo.

Reply:
I'm curious as to the reasoning behind this.

One could imagine wanting to consume the same values to (e.g.) to populate "Make" variables.